### PR TITLE
Expose cpuid and secrets data structures in the helloworld kernel.

### PR DESCRIPTION
**Pull request by andrisaar**

### CODE REVIEW

File: testing/sev_snp_hello_world_kernel/src/main.rs [hunk 1]
```diff
@@ -17,16 +17,30 @@
...
+#[link_section = ".secrets"]
+static SECRETS: MaybeUninit<SecretsPage> = MaybeUninit::uninit();
+
 #[no_mangle]
 pub extern "C" fn rust64_start() -> ! {
     serial::init_logging();
     log::info!("Hello World!");
+
+    #[allow(dead_code)]
+    let _cpuid: &CpuidPage = unsafe { CPUID.assume_init_ref() };
```
Comment (conradgrobler) [inline testing/sev_snp_hello_world_kernel/src/main.rs:39]: Please add some comments on why this unsafe code is ok.

**Review by conradgrobler — APPROVED** [review]
Thanks

File: testing/sev_snp_hello_world_kernel/src/main.rs (hunk 1 above)
Comment (tiziano88) [inline testing/sev_snp_hello_world_kernel/src/main.rs:39]: Why does it need to be assigned to a local variable? Does it get optimized away otherwise?

**Review by tiziano88 — APPROVED** [review]

File: testing/sev_snp_hello_world_kernel/src/main.rs (hunk 1 above)
Comment (andrisaar) [reply testing/sev_snp_hello_world_kernel/src/main.rs:39]: Yep, Rust optimized away the whole static, much to my surprise -- even if I tagged it with `#[used]`.

However, I've added some statements logging the contents of these pages (as we're interested what's in them anyway), so these are no longer unused.

Comment (andrisaar) [reply testing/sev_snp_hello_world_kernel/src/main.rs:39]: Done.

**Review by jul-sh — APPROVED** [review]

**Pull request merged**